src/compiler/emitter.ts

```diff
@@ -1592,13 +1592,13 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 /// these emit into an object literal property name, we don't need to be worried
                 /// about keywords, just non-identifier characters
                 function emitAttributeName(name: Identifier) {
-                    if (/[A-Za-z_]+[\w*]/.test(name.text)) {
-                        write("\"");
+                    if (/^[A-Za-z_]+[\w]*$/.test(name.text)) {

```

Comment:
1. the brackets are not needed (as we discussed in person).
2. Why have `+` after `[A-Za-z_]`? I would think that `\w*` is enough on its own to match identifier characters.
3. Does \w match "Unicode identifier-like characters"? IS there any easy way to match those characters?